### PR TITLE
initial implementation of PII checking

### PR DESCRIPTION
**Pull request by doctoryes**
https://openedx.atlassian.net/browse/PLAT-2399

Adds all necessary code to perform PII annotation checking during CI of the newly-cookie-cut IDA.

### CODE REVIEW

File: {{cookiecutter.repo_name}}/.pii_annotations.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+source_path: ./
+report_path: pii_reports
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/.pii_annotations.yml]: `pii_report` (singular)

File: {{cookiecutter.repo_name}}/.pii_annotations.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+safelist_path: .annotation_safe_list.yml
+coverage_target: 100.0
+annotations:
+    ".. no_pii::":
+    "pii_group":
+        - ".. pii::":
+        - ".. pii_types::":
+            choices: [id, name, other]
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/.pii_annotations.yml]: What do we think about the long list of pii types in OEP-30 under "The potential values of pii_types are:"?  Should we populate all those values here too?

File: {{cookiecutter.repo_name}}/openedx.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# This file describes this Open edX repo, as described in OEP-2:
+# http://open-edx-proposals.readthedocs.io/en/latest/oeps/oep-0002.html#specification
+
+nick: {{cookiecutter.repo_name[:5]}}
+oeps:
+  oep-30:
+    state: True
+openedx-release: {ref: master}
+owner: edx/core
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/openedx.yaml]: `edx/core` would be the default owner of this new repo, is there a way to make this variable?

File: {{cookiecutter.repo_name}}/.annotation_safe_list.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# This is a Code Annotations automatically-generated Django model safelist file.
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/.annotation_safe_list.yml:1]: Nice front-loading :)

File: {{cookiecutter.repo_name}}/Makefile [hunk 5]
```diff
@@ -17,6 +17,7 @@ help:
 	@echo "  pull_translations          pull translations from Transifex"
 	@echo "  push_translations          push source translation files (.po) from Transifex"
 	@echo "  quality                    run PEP8 and Pylint"
+	@echo "  pii_check                  check for PII annotations on all Django models"
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/Makefile:20]: So, no `pii_clean` target?  I was thinking the PII report cleanup could just be lumped into the `clean` target, what do you think?

File: {{cookiecutter.repo_name}}/.pii_annotations.yml (hunk 2 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/.pii_annotations.yml]: Yes - I'll do this.

File: {{cookiecutter.repo_name}}/openedx.yaml (hunk 3 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/openedx.yaml]: Sure - I'll make the cookiecutter ask about the owner.

File: {{cookiecutter.repo_name}}/Makefile (hunk 5 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/Makefile:20]: I'll add a clean-up target.

**Review by pwnage101 — APPROVED** [review]

File: {{cookiecutter.repo_name}}/setup.cfg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+[pycodestyle]
+ignore=E501
+max-line-length = 120
+exclude=.git,settings,migrations,{{cookiecutter.repo_name}}/static,bower_components,{{cookiecutter.repo_name}}/wsgi.py
```
Comment (pwnage101) [inline {{cookiecutter.repo_name}}/setup.cfg:4]: Does this mean we should also delete the .pep8 file?

File: {{cookiecutter.repo_name}}/.annotation_safe_list.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+# These models must be annotated as follows in order to be counted in the coverage report.
+# See https://code-annotations.readthedocs.io/en/latest/safelist.html for more information.
+#
+# fake_app_1.FakeModelName:
+#    ".. no_pii::": "This model has no PII"
+# fake_app_2.FakeModel2:
+#    ".. choice_annotation::": foo, bar, bazadmin.LogEntry: {}
```
Comment (bmedx) [inline {{cookiecutter.repo_name}}/.annotation_safe_list.yml]: This looks like a bug with the template safelist?

File: {{cookiecutter.repo_name}}/setup.cfg (hunk 6 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/setup.cfg:4]: It does! Meant to do that deletion and forgot - will do.

File: {{cookiecutter.repo_name}}/.annotation_safe_list.yml (hunk 7 above)
Comment (doctoryes) [reply {{cookiecutter.repo_name}}/.annotation_safe_list.yml]: That makes *much* more sense now! Yes - there's a newline missing in the safelist generation at the end of the generated comment.

**Review by bmedx — APPROVED** [review]
👍 when the outstanding feedback is addressed

**Comment by pwnage101** [comment]
Seems like you still need to delete .pep8?

**Comment by doctoryes** [comment]
I went ahead with merging this - too confusing to add pip-tools as well all at once...

**Pull request merged**